rcon/steam_utils.py

```diff
@@ -67,234 +106,297 @@ def wrapper(steamd_id: str, *args, **kwargs):
     return decorator
 
 
-def get_steam_api_key() -> str | None:
-    global last_steam_api_key_warning
+@ttl_cache(60 * 60 * 12, cache_falsy=False, is_method=False)
+@filter_steam_id()
+def fetch_steam_player_summary_player(
+    steam_id_64: str,
+) -> SteamPlayerSummaryType | None:
+    """Fetch a single players steam profile
+
+    This should never be used in any context where you know more than one steam ID
+    in advance, instead use `fetch_steam_player_summary_mult_players` and pass all
+    the steam IDs at once to reduce API calls
+    """
+    player_prof = fetch_steam_player_summary_mult_players(steam_id_64s=[steam_id_64])
+    if player_prof:
+        return player_prof.get(steam_id_64)
 
-    config = SteamUserConfig.load_from_db()
-    if config.api_key is None or config.api_key == "":
-        timestamp = datetime.datetime.now()
-        # Only log once every 5 minutes or it is super spammy
-        if (timestamp - last_steam_api_key_warning).total_seconds() > 300:
-            logger.warning("Steam API key is not set some features will be disabled.")
-            last_steam_api_key_warning = timestamp
 
-    return config.api_key
+@ttl_cache(60 * 60 * 12, cache_falsy=False, is_method=False)
+@filter_steam_ids()
+def fetch_steam_player_summary_mult_players(
+    steam_id_64s: Iterable[str], page_size=100
```

Comment:
is there a need to have the page_size as a parameter? I have trouble to find use cases, where one would _force_ to have more requests than actually needed. And it more or less sounds like an implementation details to me, which callers should not have a way to manipulate.

Reply:
I moved it to a constant in the module, it wasn't actually passed in at all anywhere it was used, it sounds like the sort of thing that will probably never change since it's in the steam API.